### ISSUE TITLE
VAE: bad convergence of loss (attached tensorboard image)

### ISSUE THREAD

**Issue by pauloabelha** [blei-lab/edward#853]
HI, I've been diving deeper into VAE and using Edward as my experimental toolbox.

I ran the exact same code as examples/vae.py, only altering one line to get tensorboard data:
Line 77: inference.initialize(optimizer=optimizer, logdir="/tmp/log")

When I check my tensorboard I see no convergence of loss function
(please see tensorboard image attached for a complete run of the code).

Final printed lower bound: -log p(x) <= 176.285
![ed_vae_tensorboard](https://user-images.githubusercontent.com/815556/36317888-324d9f9e-1336-11e8-8730-976b0247d99c.png)

Am I missing something?

Thank you,

**Comment by dustinvtran** [comment]
Nope. You are correct. Happy to take a contribution changing the number of iterations to one where it stabilizes.

**Comment by pauloabelha** [comment]
OK. Will work on it.

**Comment by pauloabelha** [comment]
To those also thinking of working on this,

I had to stop the running after two days. I need a better computer.
I got to:
-log p(x) <= 207.419
Epoch: 17768

See attached image for tensorboard.
![ed_vae_tensorboard_20k](https://user-images.githubusercontent.com/815556/36372317-9b8e3428-155c-11e8-85ea-f8d371d43e0d.png)

**Comment by dustinvtran** [comment]
It seems unnecessary to be super-specific about the number of training iterations if it takes that long. Maybe try larger batch sizes (e.g., 256) and a non-trivial latent dimension (e.g., 50)?

**Comment by pauloabelha** [comment]
Yes, thank you. I’ll experiment. I just had to stop for a bit and decided to dump this result here for anyone interested in the issue.
